### PR TITLE
3.4 - Response migration docs

### PR DESCRIPTION
**Pull request by markstory**
Start the PSR7 Response migration guide changes. I'll update the response chapter separately as it will be a big set of changes.

Refs cakephp/cakephp#9636

### CODE REVIEW

File: en/appendices/3-4-migration-guide.rst [hunk 1]
```diff
@@ -71,6 +123,64 @@ Cake\Validation\Validator
 Cake\View\StringTemplateTrait
     * ``templates()``
 
+.. _adopting-immutable-responses:
+
+Adopting Immutable Responses
+============================
+
+Before you migrate your code to use the new response methods you should be aware
+of the semantic changes the new methods create. The immutable methods are
```
Comment (markstory) [inline en/appendices/3-4-migration-guide.rst]: Conceptual?

Comment (HavokInspiration) [inline en/appendices/3-4-migration-guide.rst]: "conceptual" have more meaning to me than "semantic".

**Comment by markstory** [comment]
Fixed up based on the feedback so far.

**Comment by lorenzo** [comment]
This looks good to me

**Pull request merged**